script/model/forecast.py

```diff
@@ -0,0 +1,585 @@
+
+        with torch.no_grad():
+            model.hidden = (torch.zeros(1, 1, model.hidden_layer_size),
+                            torch.zeros(1, 1, model.hidden_layer_size))
+            pred = model(seq)
+
+        if self.y_transformer:
+            pred = self.y_transformer.inverse_transform(pred.reshape(1, -1))
+
+        return pred.item()
+
+
+if __name__ == "__main__":
+    args = argp.parse_args()
+
+    if args.test_file is None:
+
+        # Generate OLTP trace file
+        if args.gen_data:
+            gen_oltp_trace(
+                tpcc_weight=args.tpcc_weight,
+                tpcc_rates=args.tpcc_rates,
+                pattern_iter=args.pattern_iter)
+
+            trace_file = DEFAULT_QUERY_TRACE_FILE
+        else:
+            trace_file = args.trace_file
+
+        data_loader = DataLoader(
+            query_trace_file=trace_file,
+            interval_us=INTERVAL_MICRO_SEC)
+        x_transformer, y_transformer = data_loader.get_transformers()
+
+        # FIXME: Assuming all the queries in the current trace file are from
+        # the same cluster for now
+        cluster_trace = QueryCluster(data_loader.get_ts_data())
+
+        # Aggregated time-series from the cluster
+        train_timeseries = cluster_trace.get_timeseries()
+
+        # Perform training on the trace file
+        lstm_model = LSTM(
+            lr=args.lr,
+            epochs=args.epochs,
+            x_transformer=x_transformer,
+            y_transformer=y_transformer)
+        trainer = ForecastTrainer(
+            train_timeseries,
+            seq_len=args.seq_len,
+            eval_size=args.eval_size,
+            horizon_len=args.horizon_len)
+        trainer.train([lstm_model])
+
+        # Save the model
+        if args.model_save_path:
+            with open(args.model_save_path, "wb") as f:
+                pickle.dump(lstm_model, f)
+    else:
+        # Do inference on a trained model
+        data_loader = DataLoader(
+            query_trace_file=args.test_file,
+            interval_us=INTERVAL_MICRO_SEC)
+
+        with open(args.model_load_path, "rb") as f:
+            model = pickle.load(f)
+        x_transformer, y_transformer = data_loader.get_transformers()
+        model.x_transformer = x_transformer
+        model.y_transformer = y_transformer
+
+        # FIXME: Assuming all the queries in the current trace file are from
+        # the same cluster for now
+        timeseries = data_loader.get_ts_data()
+        cluster_trace = QueryCluster(timeseries)
+
+        # Aggregated time-series from the cluster
+        test_timeseries = cluster_trace.get_timeseries()
+
+        pred = model.test(test_timeseries[:args.seq_len])
+
+        query_pred = cluster_trace.segregate([pred])
+        for qid, ts in query_pred.items():
+            LOG.info(
+                f"[Query: {qid} @idx={args.seq_len+args.horizon_len}] "
+                f"actual={timeseries[qid][args.seq_len+args.horizon_len]},pred={ts[0]:.1f}")
```

Comment:
Move these also inside `Forecaster`, e.g., inside the `eval` method. We should also directly load the transformers from the model dump after we move the transforms inside the model.